cluster-autoscaler/cloudprovider/clusterapi/clusterapi_unstructured.go

```diff
@@ -0,0 +1,216 @@
+	return r.unstructured.GetNamespace()
+}
+
+func (r unstructuredScalableResource) ProviderIDs() ([]string, error) {
+	providerIds, err := r.controller.scalableResourceProviderIDs(r.unstructured)
+	if err != nil {
+		return nil, err
+	}
+
+	return providerIds, nil
+}
+
+func (r unstructuredScalableResource) Replicas() (int32, error) {
+	gvr, err := r.GroupVersionResource()
+	if err != nil {
+		return 0, err
+	}
+
+	s, err := r.controller.managementScaleClient.Scales(r.Namespace()).Get(context.TODO(), gvr.GroupResource(), r.Name(), metav1.GetOptions{})
+	if err != nil {
+		return 0, err
+	}
+	if s == nil {
+		return 0, fmt.Errorf("unknown %s %s/%s", r.Kind(), r.Namespace(), r.Name())
+	}
+	return s.Spec.Replicas, nil
+}
+
+func (r unstructuredScalableResource) SetSize(nreplicas int32) error {
+	gvr, err := r.GroupVersionResource()
+	if err != nil {
+		return err
+	}
+
+	s, err := r.controller.managementScaleClient.Scales(r.Namespace()).Get(context.TODO(), gvr.GroupResource(), r.Name(), metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if s == nil {
+		return fmt.Errorf("unknown %s %s/%s", r.Kind(), r.Namespace(), r.Name())
+	}
+
+	s.Spec.Replicas = nreplicas
+	_, updateErr := r.controller.managementScaleClient.Scales(r.Namespace()).Update(context.TODO(), gvr.GroupResource(), s, metav1.UpdateOptions{})
+	return updateErr
+}
+
+func (r unstructuredScalableResource) UnmarkMachineForDeletion(machine *unstructured.Unstructured) error {
+	u, err := r.controller.managementClient.Resource(r.controller.machineResource).Namespace(machine.GetNamespace()).Get(context.TODO(), machine.GetName(), metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	annotations := u.GetAnnotations()
+	if _, ok := annotations[machineDeleteAnnotationKey]; ok {
+		delete(annotations, machineDeleteAnnotationKey)
+		u.SetAnnotations(annotations)
+		_, updateErr := r.controller.managementClient.Resource(r.controller.machineResource).Namespace(u.GetNamespace()).Update(context.TODO(), u, metav1.UpdateOptions{})
+
+		return updateErr
+	}
+
+	return nil
+}
+
+func (r unstructuredScalableResource) MarkMachineForDeletion(machine *unstructured.Unstructured) error {
+	u, err := r.controller.managementClient.Resource(r.controller.machineResource).Namespace(machine.GetNamespace()).Get(context.TODO(), machine.GetName(), metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	u = u.DeepCopy()
+
+	annotations := u.GetAnnotations()
+	if annotations == nil {
+		annotations = map[string]string{}
+	}
+
+	annotations[machineDeleteAnnotationKey] = time.Now().String()
+	u.SetAnnotations(annotations)
+
+	_, updateErr := r.controller.managementClient.Resource(r.controller.machineResource).Namespace(u.GetNamespace()).Update(context.TODO(), u, metav1.UpdateOptions{})
+
+	return updateErr
+}
+
+func (r unstructuredScalableResource) Labels() map[string]string {
+	labels, found, err := unstructured.NestedStringMap(r.unstructured.Object, "spec", "template", "spec", "labels")
```

Comment:
I think this should be this?
```suggestion
	labels, found, err := unstructured.NestedStringMap(r.unstructured.Object, "spec", "template", "spec", "metadata", "labels")
```

Reply:
lol, funny. i was just changing this XD